packages/block-library/src/quote/v2/index.js

```diff
@@ -35,3 +31,46 @@ const settings = {
 };
 
 export default settings;
+
+/**
+ * This function updates the attributes for the quote v2.
+ * This should be moved to block.json when v2 becomes the default.
+ *
+ * @param {Array}  blockSettings The settings of the block to be registered.
+ * @param {string} blockName     The name of the block to be registered.
+ * @return {Array} New settings.
+ */
+function registerQuoteV2Attributes( blockSettings, blockName ) {
+	if ( ! window?.__experimentalEnableQuoteBlockV2 ) {
+		return blockSettings;
+	}
+
+	if ( blockName !== 'core/quote' ) {
+		return blockSettings;
+	}
+
+	// Register the new attribute.
+	Object.assign( blockSettings.attributes, {
+		attribution: {
+			type: 'string',
+			source: 'html',
+			selector: 'figcaption',
+			default: '',
+			__experimentalRole: 'content',
+		},
```

Comment:
What would be the issue if we use the same block.json file for both versions. I mean aside having all the old attributes in the version, which doesn't bother me much specially since we need to keep them to implement the "migration effect" behavior.

Reply:
It was the other way around I was concerned about: I didn't want to pollute the existing quote block with the v2 attributes.

Technically, it might work in this case, as we're not _modifying_ existing atts but _adding_ new ones. I can't think of anything that would break. Though I still not a fan of this approach, I rather keep things separated.

Reply:
Ok, let's keep them separated for now, it doesn't hurt either.